classes/plugins/PluginHelper.inc.php

```diff
@@ -256,22 +267,30 @@ function _cutAndPasteTemporaryFile($path, $pluginDest, $pluginLibDest, &$errorMs
 		// Start with the library part (if any).
 		$libPath = $path . '/lib';
 		if (is_dir($libPath)) {
-			if(!$fileManager->copyDir($libPath, $pluginLibDest)) {
+			if (!$this->_cutAndPasteDir($fileManager, $libPath, $pluginLibDest)) {
 				$errorMsg = __('manager.plugins.copyError');
-				return null;
+				return;
 			}
-			// Remove the library part of the temporary folder.
-			$fileManager->rmtree($libPath);
 		}
 
-		// Continue with the application-specific part (mandatory).
-		if(!$fileManager->copyDir($path, $pluginDest)) {
+		// Continue with the application-specific part (mandatory)
+		if (!$this->_cutAndPasteDir($fileManager, $path, $pluginDest)) {
 			$errorMsg = __('manager.plugins.copyError');
-			return null;
+			return;
 		}
+	}
 
-		// Remove the temporary folder.
-		$fileManager->rmtree(dirname($path));
+	/**
+	 * Cuts and pastes source to target.
+	 * @param $manager The file manager to use
+	 * @param $src The source directory
+	 * @param $trg The target directory
+	 * @return Was it successful
```

Comment:
The types are missing for `@param` and `@return` variables -- e.g. `@param $manager FileManager`. Also for the function `_cutAndPasteTemporaryFile` above.